ferrocene/ci/scripts/mirror-sphinx-shared-resources.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: MIT OR Apache-2.0
+# SPDX-FileCopyrightText: The Ferrocene Developers
+
+
+import os
+import random
+import shlex
+import string
+import subprocess
+import tempfile
+
+
+COMMIT_MSG = "Automated push to `ferrocene/sphinx-shared-resources`"
+MIRROR_DIR = "ferrocene/doc/sphinx-shared-resources"
+MIRROR_REPO = "git@github.com:ferrocene/sphinx-shared-resources.git"
+
+
+# Mirror changes to ferrocene/sphinx-shared-resources.
+#
+# How:
+# 1. Clone ferrocene/sphinx-shared-resources.
+# 2. Copy files from ferrocene/ferrocene subdirectory to cloned repo.
+# 3. Commit and push changes, if there are any.
+#
+# Note that all commands are executed in a temporary directory.
+def mirror_changes():
+    cwd = os.getcwd()
+
+    # clone mirror repository into temporary directory
+    run(f"git clone {MIRROR_REPO} .")
+
+    # create branch
+    rand_id = "".join(random.choices(string.ascii_lowercase + string.digits, k=8))
+    branch_name = f"automation/mirror-sphinx-shared-resources/{rand_id}"
+    run(f"git switch -c {branch_name}")
+
+    # copy contents of mirror directory into mirror repository
+    # `-T` is responsible for only copying the content
+    run(f"cp -frT {cwd}/{MIRROR_DIR} .")
+
+    # commit changes
+    run("git add .")
+    returncode = run(f'git commit -m "{COMMIT_MSG}"', check=False)
+    new_commit = returncode == 0
+
+    if not new_commit:
+        print("No changes")
+        return
+
+    # push the commit
+    run(f"git push -u origin {branch_name}")
+
+    # create pull request
+    run(f"gh pr create --body {""} --title \"{COMMIT_MSG}\" --head \"{branch_name}\" --base main")
```

Comment:
I would not push and create the PR in the script itself, as that hinders testing stuff locally. If the script finishes at creating the commit you can run it however many times locally while testing without worrying it messes up the remote repository :slightly_smiling_face: